temporal-serviceclient/src/main/java/io/temporal/serviceclient/WorkflowServiceStubsOptions.java

```diff
@@ -445,10 +446,19 @@ public Builder setRpcTimeout(Duration timeout) {
 
     /**
      * Sets the rpc timeout value for the following long poll based operations:
-     * PollWorkflowTaskQueue, PollActivityTaskQueue, GetWorkflowExecutionHistory. Should never be
-     * below 60 seconds as this is server side timeout for the long poll. Default is 70 seconds.
+     * PollWorkflowTaskQueue, PollActivityTaskQueue, GetWorkflowExecutionHistory.
+     *
+     * <p>Server side timeout for the long poll is 60s. This parameter should never be below 70
+     * seconds (server timeout + additional delay). Default is 70 seconds.
+     *
+     * @throws IllegalArgumentException if {@code timeout} is less than 70s
+     * @deprecated exposing of this option for users configuration deemed non-beneficial and
+     *     dangerous
      */
+    @Deprecated
     public Builder setRpcLongPollTimeout(Duration timeout) {
+      Preconditions.checkArgument(
+          timeout.toMillis() > 70_000, "rpcLongPollTimeout has to be longer 70s");
```

Comment:
Since we're already deprecating here I dunno that we need to potentially cause people pain by throwing here if they're already setting it too low.

I can go either way here though so if you want to keep it go for it.

Reply:
Setting less than 70s here now is just dangerous. So, I think it's the best we can do here.